src/bin/good-fences.rs

```diff
@@ -1,32 +1,106 @@
 extern crate good_fences_rs_core; // Optional in Rust 2018
 extern crate serde_json;
+use clap::{self, Parser};
+use good_fences_rs_core::evaluate_fences::ImportRuleViolation;
 use good_fences_rs_core::good_fences_runner::GoodFencesRunner;
-use std::env::set_current_dir;
+use serde::Serialize;
+use std::env::{self, set_current_dir};
 use std::path::Path;
 use std::time::Instant;
 
+#[derive(Debug, Parser)]
+struct Cli {
+    /**
+     * Dirs to look for fence and source files
+     */
+    paths: Vec<String>,
+
+    /**
+     * The tsconfig file used relative to '--root' argument
+     */
+    #[clap(short, long, default_value = "tsconfig.json")]
+    project: String,
+
+    /**
+     *  Overrides `compilerOptions.baseUrl` property read from '--project' argument
+     */
+    #[clap(short, long)]
+    base_url: Option<String>,
+
+    /**
+     * Argument to change the cwd of execution
+     */
+    #[clap(short, long, default_value = ".")]
+    root: String,
+
+    /**
+     * Output file for violations, relative to '--root' argument
+     */
+    #[clap(short, long, default_value = "good-fences-violations.json")]
+    output: String,
+}
+
 fn main() {
     // set working dir
     let start = Instant::now();
-    let root = Path::new("../client-web");
-    // let mut file = File::create("./good-fences-errors.log").unwrap();
+    let args = Cli::parse();
+    let root = Path::new(args.root.as_str());
+
     assert!(set_current_dir(&root).is_ok());
     println!(
         "Successfully changed working directory to {}!",
         root.display()
     );
 
     println!("beginning file walks");
-    let good_fences_runner =
-        GoodFencesRunner::new("tsconfig.paths.json", &vec!["packages", "shared"]);
-    // println!("{:#?}", good_fences_runner);
+    let tsconfig_path = args.project;
+    let dirs_to_walk = &args.paths.iter().map(|x| x.as_str()).collect();
+    let good_fences_runner = GoodFencesRunner::new(&tsconfig_path, dirs_to_walk);
```

Comment:
this should be constructed before we call `root`, since otherwise the `--project` will be evaluated relative to the `--root`.

I think generally that's considered bad form when making a CLI.

Reply:
If you want to have `base_url` as a cli argument, You should update GoodfencesRunner::new to accept the already parsed tsconfig.json file, and then override the base_url of the tsconfig_paths json if it is unset

Reply:
Can you check on how was this addressed?